src/Nethermind/Nethermind.Shutter.Test/ShutterBlockHandlerTests.cs

```diff
@@ -0,0 +1,98 @@
+// SPDX-FileCopyrightText: 2024 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using NUnit.Framework;
+using Nethermind.Core.Test.Builders;
+using System.Threading;
+using System.Threading.Tasks;
+using Nethermind.Core;
+using System;
+
+namespace Nethermind.Shutter.Test;
+
+[TestFixture]
+class ShutterBlockHandlerTests
+{
+    [Test]
+    public async Task Can_wait_for_valid_block()
+    {
+        Random rnd = new(ShutterTestsCommon.Seed);
+        Timestamper timestamper = ShutterTestsCommon.InitTimestamper(ShutterTestsCommon.InitialSlotTimestamp, 0);
+
+        ShutterApiSimulator api = ShutterTestsCommon.InitApi(rnd, timestamper);
+        IShutterBlockHandler blockHandler = api.BlockHandler;
+
+        bool waitReturned = false;
+        CancellationTokenSource source = new();
+        _ = blockHandler.WaitForBlockInSlot(ShutterTestsCommon.InitialSlot, source.Token)
+            .ContinueWith((_) => waitReturned = true)
+            .WaitAsync(source.Token);
+
+        await Task.Delay((int)(api.BlockWaitCutoff.TotalMilliseconds / 2));
+
+        Assert.That(waitReturned, Is.False);
+
+        api.TriggerNewHeadBlock(new(Build.A.Block.WithTimestamp(ShutterTestsCommon.InitialSlotTimestamp).TestObject));
+
+        await Task.Delay(100);
+        Assert.That(waitReturned);
+    }
```

Comment:
This is a simplified version of this test, that still depends on timing from the `CancellationTokenSource` internally, but at least the test itself doesn't use Task.Delay.
```suggestion
    [Test]
    public void Can_wait_for_valid_block()
    {
        Random rnd = new(ShutterTestsCommon.Seed);
        Timestamper timestamper = ShutterTestsCommon.InitTimestamper(ShutterTestsCommon.InitialSlotTimestamp, 0);

        ShutterApiSimulator api = ShutterTestsCommon.InitApi(rnd, timestamper);
        IShutterBlockHandler blockHandler = api.BlockHandler;

        CancellationTokenSource source = new();
        Task<Block?> waitTask = blockHandler.WaitForBlockInSlot(ShutterTestsCommon.InitialSlot, source.Token);

        Block result = Build.A.Block.WithTimestamp(ShutterTestsCommon.InitialSlotTimestamp).TestObject;
        api.TriggerNewHeadBlock(new(result));

        Assert.That(result, Is.EqualTo(waitTask.Result));
    }
```